src/controllers/expenseController.js

```diff
@@ -0,0 +1,102 @@
+const {
+  filterExpenses,
+  addExpense,
+  getExpense,
+  deleteExpense,
+  updateExpense,
+  getExpenses,
+} = require('../services/expenses.service');
+const { getUser } = require('../services/users.service');
+
+const getAll = (req, res) => {
+  const { userId, categories, from, to } = req.query;
+
+  if (userId || categories || from || to) {
+    res.statusCode = 200;
+    res.send(filterExpenses(req.query));
+
+    return;
+  }
+
+  const expenses = getExpenses();
+
+  res.statusCode = 200;
+  res.send(expenses);
+};
+
+const add = (req, res) => {
+  const { userId, category, spentAt, title, amount, note } = req.body;
+  const user = getUser(userId);
+
+  if (!user || !category || !spentAt || !title || !amount) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const rawExpense = {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
+
+  const expense = addExpense(rawExpense);
+
+  res.statusCode = 201;
+  res.send(expense);
+};
+
+const get = (req, res) => {
+  const { id } = req.params;
+  const expense = getExpense(+id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.statusCode = 200;
+  res.send(expense);
+};
+
+const remove = (req, res) => {
+  const { id } = req.params;
+  const expense = getExpense(+id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  deleteExpense(+id);
+  res.sendStatus(204);
+};
+
+const update = (req, res) => {
+  const { id } = req.params;
+  const expense = getExpense(+id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  const newExpense = Object.assign(expense, req.body);
```

Comment:
Using `Object.assign` to merge the existing expense with the new data from `req.body` is fine, but be cautious about overwriting fields that should not be changed. Consider explicitly listing the fields that are allowed to be updated.